homeassistant/components/reolink/binary_sensor.py

```diff
@@ -0,0 +1,172 @@
+    icon_off: str = "mdi:motion-sensor-off"
+    value: Callable[[ReolinkHost, int | None], bool | None] = lambda host, ch: None
+    supported: Callable[[ReolinkHost, int | None], bool] = lambda host, ch: True
+
+
+BINARY_SENSORS = (
+    ReolinkBinarySensorDescription(
+        key="motion",
+        name="Motion",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.motion_detected(ch),
+    ),
+    ReolinkBinarySensorDescription(
+        key=FACE_DETECTION_TYPE,
+        name=FACE_DETECTION_TYPE,
+        icon="mdi:face-recognition",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.ai_detected(ch, FACE_DETECTION_TYPE),
+        supported=lambda host, ch: host.api.ai_supported(ch, FACE_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorDescription(
+        key=PERSON_DETECTION_TYPE,
+        name=PERSON_DETECTION_TYPE,
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.ai_detected(ch, PERSON_DETECTION_TYPE),
+        supported=lambda host, ch: host.api.ai_supported(ch, PERSON_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorDescription(
+        key=VEHICLE_DETECTION_TYPE,
+        name=VEHICLE_DETECTION_TYPE,
+        icon="mdi:car",
+        icon_off="mdi:car-off",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.ai_detected(ch, VEHICLE_DETECTION_TYPE),
+        supported=lambda host, ch: host.api.ai_supported(ch, VEHICLE_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorDescription(
+        key=PET_DETECTION_TYPE,
+        name=PET_DETECTION_TYPE,
+        icon="mdi:dog-side",
+        icon_off="mdi:dog-side-off",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.ai_detected(ch, PET_DETECTION_TYPE),
+        supported=lambda host, ch: host.api.ai_supported(ch, PET_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorDescription(
+        key="visitor",
+        name="Visitor",
+        icon="mdi:bell-ring-outline",
+        icon_off="mdi:doorbell",
+        value=lambda host, ch: host.api.visitor_detected(ch),
+        supported=lambda host, ch: host.api.is_doorbell_enabled(ch),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Reolink IP Camera."""
+    reolink_data: ReolinkData = hass.data[DOMAIN][config_entry.entry_id]
+    host = reolink_data.host
+
+    entities = []
+    for channel in host.api.channels:
+        for entity_description in BINARY_SENSORS:
+            if not entity_description.supported(host, channel):
+                continue
+
+            entities.append(
+                ReolinkBinarySensorEntity(
+                    reolink_data,
+                    channel,
+                    entity_description,
+                )
+            )
+
+    async_add_entities(entities)
```

Comment:
Bit of personal styling, but the `host` variable seems unneeded and a list comprehension can be used in the middle part without being harmful to readability.

I would write it as:

```suggestion
    reolink_data: ReolinkData = hass.data[DOMAIN][config_entry.entry_id]

    entities: list[ReolinkBinarySensorEntity] = []
    for channel in reolink_data.host.api.channels:
        entities.extend(
            ReolinkBinarySensorEntity(
                reolink_data,
                channel,
                entity_description,
            )
            for entity_description in BINARY_SENSORS
            if entity_description.supported(reolink_data.host.api, channel)
        )
    async_add_entities(entities)
```

Reply:
Thanks, that makes it more compact.